x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/step_configure_datasource.tsx

```diff
@@ -5,28 +5,29 @@
  */
 import React from 'react';
 import { FormattedMessage } from '@kbn/i18n/react';
-import {
-  EuiPanel,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiSpacer,
-  EuiEmptyPrompt,
-  EuiText,
-  EuiCallOut,
-} from '@elastic/eui';
+import { EuiPanel, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiCallOut } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { PackageInfo, NewDatasource, DatasourceInput } from '../../../types';
 import { Loading } from '../../../components';
 import { DatasourceValidationResults, validationHasErrors } from './services';
-import { DatasourceInputPanel } from './components';
+import { DatasourceInputPanel, CustomConfigureDatasource } from './components';
+import { CreateDatasourceFrom } from './types';
 
 export const StepConfigureDatasource: React.FunctionComponent<{
+  from?: CreateDatasourceFrom;
   packageInfo: PackageInfo;
-  datasource: NewDatasource;
+  datasource: NewDatasource | (NewDatasource & { id: string });
```

Comment:
maybe? 😬 

If also export the `CustomConfigureDatasourceProps`, then you can:

```suggestion
  datasource:   CustomConfigureDatasourceProps['datasource'];
```